diff --git a/www/js/app.js b/www/js/app.js
index eaf2e5e..101b678 100644
--- a/www/js/app.js
+++ b/www/js/app.js
@@ -11,7 +11,20 @@
 
 window.app = window.app || {} ;         // don't clobber existing app object
 
-
+/*var app.timer = {}
+var app.getDataFromServer = function(){
+    var xhr = new XMLHttpRequest();
+    xhr.open('GET', "http://localhost:5000/todo/api/v1.0/tasks", true);
+    xhr.send();
+    xhr.onreadystatechange = processRequest;
+    function processRequest(e) {
+        if (xhr.readyState == 4 && xhr.status == 200)
+        {
+            var response = JSON.parse(xhr.responseText);
+            app.consoleLog(xhr.responseText.toString()) ;
+        }
+    }   
+}*/
 // The console.log() messages sprinkled in this file are for instruction and debug.
 // If you reuse this code you do not need to include them as part of your app.
 // Set to "true" if you want the console.log messages to appear.
@@ -41,6 +54,8 @@ app.btnBeep = function() {
     }
 
     app.consoleLog(fName, "exit") ;
+    app.timer = setInterval;
+    
 } ;
 
 
@@ -59,6 +74,7 @@ app.btnVibrate = function() {
     }
 
     app.consoleLog(fName, "exit") ;
+    clearInterval(timer);
 } ;
 
 app.rotateMatrixX = {};
@@ -89,7 +105,10 @@ app.btnGyro = function(){
         var _gamma = Math.round(gyro.gamma);
         
         document.getElementById('gyro').value = _alpha + ", " + _beta + ", " + _gamma;
-        
+                    var str = write("gyroscope.output", getDateToStr() + "," +
+                                        _alpha + "," +
+                                        _beta + "," +
+                                        _gamma);
         if (app.Fix == "yes") 
         {
             app.Fix = "no";
@@ -117,10 +136,6 @@ app.btnGyro = function(){
                                                 [0,         0,          1] ]); // Matrix
             
             app.rotateMatrix = math.multiply(app.rotateMatrixZ, math.multiply(app.rotateMatrixX, app.rotateMatrixY));  
-            var str = write("gyroscope.output", getDateToStr() + "," +
-                                        _alpha + "," +
-                                        _beta + "," +
-                                        _gamma);
         }
     }
 
diff --git a/www/js/init-app.js b/www/js/init-app.js
index 1abf0be..ec782fa 100644
--- a/www/js/init-app.js
+++ b/www/js/init-app.js
@@ -57,6 +57,7 @@ app.init.events = function() {
     acc.initCompass() ;
     geo.initGeoLocate() ;
     app.updateDeviceInfo() ;
+    
 
     // NOTE: initialize your app event handlers
     // See main.js, cordova-acc.js and cordova-geo.js for event handlers.
